kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -182,7 +183,9 @@ public Transaction build(Engine engine) {
       newMetadata = Optional.of(snapshotMetadata.withMergedConfiguration(newProperties));
     }
 
-    // TODO In the future update metadata with new schema if provided
+    if (schema.isPresent() && !isNewTable) {
```

Comment:
did we add a test that this re-ordering means we correctly assign the field-ids/physical names to new columns?